userspace/Build.mk

```diff
@@ -130,9 +130,16 @@ build/userspace/$(APP)/$(BOARD)/full_image: \
 	arm-none-eabi-objcopy --update-section \
 		.apps=build/userspace/$(APP)/cortex-m3/cortex-m3.tbf \
 		build/userspace/$(APP)/$(BOARD)/unsigned_image
-	$(TANGO_CODESIGNER) --b --input build/userspace/$(APP)/$(BOARD)/unsigned_image \
-		--key=$(TANGO_CODESIGNER_KEY) \
-		--output=build/userspace/$(APP)/$(BOARD)/signed_image
+	if [ -n "${TANGO_CODESIGNER}" -a -n "${TANGO_CODESIGNER_KEY}" ]; then \
+		$(TANGO_CODESIGNER) --b --input build/userspace/$(APP)/$(BOARD)/unsigned_image \
+			--key=$(TANGO_CODESIGNER_KEY) \
+			--output=build/userspace/$(APP)/$(BOARD)/signed_image; \
+	else \
+		echo "***** Substituting unsigned_image for signed_image -- one of"; \
+		echo "*****   CODESIGNER{,_KEY} is empty-- hope that's OK."; \
+		cp -a build/userspace/$(APP)/$(BOARD)/unsigned_image \
+			build/userspace/$(APP)/$(BOARD)/signed_image; \
+	fi
 	cat $(TANGO_BOOTLOADER) build/userspace/$(APP)/$(BOARD)/signed_image \
```

Comment:
I think you should pull the `cat ... > .../full_image` into the `if`-clause, as well. Pretending to generate a signed_image and with that the full_image is confusing.

Reply:
Hmm.  OK.  For some reason, I had convinced myself that `make build` would fail when `full_image` isn't created, but I guess I never tested it.  There _are_ targets that depend on `full_image`'s existence, but I guess `make build` doesn't need them.  So later steps will fail if we don't get a `full_image`, but I guess that's OK.

Reply:
Done.